common/src/main/java/org/opensearch/ml/common/connector/ConnectorAction.java

```diff
@@ -185,6 +198,90 @@ public static ConnectorAction parse(XContentParser parser) throws IOException {
             .build();
     }
 
+    public void validatePrePostProcessFunctions(Map<String, String> parameters) {
+        StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
+        String endPoint = substitutor.replace(url);
+        String remoteServer = getRemoteServerFromURL(endPoint);
+        validatePreProcessFunctions(remoteServer);
```

Comment:
if (isInBuiltProcessFunction(preProcessFunction)) then we can invoke validatePreProcessFunctions?

Reply:
corrected.